File: java/src/test/java/ConvertIntegerToWordedStringTests.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import lib.Converter;

public class ConvertIntegerToWordedStringTests {

    @Test
    // TODO: This test will become obsolete if the featured function is fully
    // implemented.
    void notImplementedYet() {
        assertThrows(UnsupportedOperationException.class, () -> {
            Converter.integerToWordedString(Integer.MIN_VALUE);
        });
    }

    @Test
    // TODO: Initial "red" test to code logic
    void firstPostiveIntegerIsOne() {
        int number = 1;
        String words = Converter.integerToWordedString(number);
        assertEquals("one", words);
    }
}

File: python/sample-tests/ConvertIntegerToWordedStringTests.py
import unittest
import sys

sys.path.append("../")  # pythonian import uniqueness
from sample.lib.Converter import IntegerToWordedString


class ConvertIntegerToWordedStringTests(unittest.TestCase):

    # TODO: This test will become obsolete once the IntegerToWordedString
    # function is fully implemented as per the specifications.
    def test_not_implemented_yet(self):
        with self.assertRaises(NotImplementedError):
            IntegerToWordedString(-1000000)

    # TODO: Initial "red" test to code logic
    def test_first_positive_integer_is_one(self):
        number = 1
        words = IntegerToWordedString(number)
        self.assertEqual(words.lower(), "one")


if __name__ == "__main__":
    unittest.main()
